pkg/authn/kubernetes/go.mod

```diff
@@ -4,9 +4,9 @@ go 1.17
 
 require (
 	github.com/google/go-containerregistry v0.8.0
-	k8s.io/api v0.22.5
-	k8s.io/apimachinery v0.22.5
-	k8s.io/client-go v0.22.5
+	k8s.io/api v0.23.2
+	k8s.io/apimachinery v0.23.2
+	k8s.io/client-go v0.23.2
```

Comment:
Didn't we explicitly leave these at 0.22?

cc @dprotaso

Reply:
> Didn't we explicitly leave these at 0.22?

@mattmoor -- There was a request to use 0.22.5, but there wasn't a reason provided:
- https://github.com/google/go-containerregistry/pull/1234#discussion_r780449706
- https://github.com/google/go-containerregistry/pull/1234#discussion_r780449910

Would suggest using 0.23.**3**, unless there's a good reason not to.

Reply:
tl;dr It's because that's what Knative and Tekton are pinned to.  I'll defer to @dprotaso to explain the cadence.

Reply:
Couldn't Knative and Tekton solve this with a `replace k8s.io/foo => k8s.io/foo v0.22.5`?

Reply:
They could, based on my understanding.
I think it's important that the upstream take input from downstreams, but set the pace by keeping deps as up-to-date as possible.